tests/Doctrine/Tests/Common/Annotations/DocParserTest.php

```diff
@@ -225,6 +226,123 @@ public function testTypicalMethodDocBlock(): void
         self::assertInstanceOf(Marker::class, $marker);
     }
 
+    public function testAnnotationWithConstructorWithVariadicParam(): void
+    {
+        $parser = $this->createTestParser();
+
+        // Without variadic arguments
+        $docblock = <<<'DOCBLOCK'
+/**
+ * @SomeAnnotationWithConstructorWithVariadicParam("Some data")
+ */
+DOCBLOCK;
+
+        $result = $parser->parse($docblock);
+        self::assertCount(1, $result);
+        $annot = $result[0];
+
+        self::assertInstanceOf(SomeAnnotationWithConstructorWithVariadicParam::class, $annot);
+
+        self::assertSame('Some data', $annot->name);
+        self::assertSame([], $annot->data);
+
+        // With named arguments
+        $docblock = <<<'DOCBLOCK'
+/**
+ * @SomeAnnotationWithConstructorWithVariadicParam(name = "Some data", foo = "Foo", bar = "Bar")
+ */
+DOCBLOCK;
+
+        $result = $parser->parse($docblock);
+        self::assertCount(1, $result);
+        $annot = $result[0];
+
+        self::assertInstanceOf(SomeAnnotationWithConstructorWithVariadicParam::class, $annot);
+
+        self::assertSame('Some data', $annot->name);
+        if (PHP_VERSION_ID >= 80000) {
+            self::assertSame(['foo' => 'Foo', 'bar' => 'Bar'], $annot->data);
+        } else {
+            self::assertSame(['Foo', 'Bar'], $annot->data);
+        }
```

Comment:
No go. The library must not change its behavior when upgrading PHP.

Reply:
It's relevant to PHP behavior. I mean spread operator. And all existing tests are passed.

Ok. If the PHP behavior is not preffered (PHP behavior would be better for migration from annotations to attributes any way), let's define the needed behavior.

```php
/** @annotatnin @NamedArgumentConstructor */
class Ann {
    public function __construct(string $foo, string ..$bar) { ... }
}
```

Just explain me thise cases and i'll try to code a solution:

1. ~~There mustn't be any exception when we just add a variadic parameter in an existing annotation~~
2. Named case 1: `Ann(foo = "foo", bar = "bar")`. $bar = ?
3. Named case 2: `Ann(foo = "foo", bar = {"bar", "baz"})`.  Error?
4. Positional case 1: `Ann("foo", "bar")`.  $bar = ?
5. Positional case 2: `Ann("foo", "bar", "baz")`.  $bar = ?
6. Will we have any ability to pass many values to $bar?

Reply:
> 6. Will we have any ability to pass many values to $bar?

My preferred solution would be to not allow to pass anything to `$bar` at all.